airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -84,23 +91,23 @@ def __post_init__(self, parameters: Mapping[str, Any]) -> None:
             else datetime.timedelta.max
         )
         self._cursor_granularity = self._parse_timedelta(self.cursor_granularity)
-        self.cursor_field = InterpolatedString.create(self.cursor_field, parameters=parameters)
-        self.lookback_window = InterpolatedString.create(self.lookback_window, parameters=parameters)
-        self.partition_field_start = InterpolatedString.create(self.partition_field_start or "start_time", parameters=parameters)
-        self.partition_field_end = InterpolatedString.create(self.partition_field_end or "end_time", parameters=parameters)
+        self._cursor_field = InterpolatedString.create(self.cursor_field, parameters=parameters)
```

Comment:
This seems like a breaking change. I understand the want to make more things private but can we confirm this won't have negative impact at least in our repo. Else, is this necessary for this PR?

Reply:
The class should only be used by a retriever and the only implementation in our codebase in the SimpleRetriever.

Context for the change was satisfying mypy. The alternative if we don't want to start using a private variable is to add `# type: ignore` in the file. I can do that if it feels safer

Reply:
Thanks for the context! I'm fine with this being breaking and we move one step in the right direction